src/api/time/TimeContext.js

```diff
@@ -353,25 +365,298 @@ class TimeContext extends EventEmitter {
       return;
     }
 
-    const newBounds = {
-      start: timestamp + this.offsets.start,
-      end: timestamp + this.offsets.end
-    };
+    if (this.mode === REALTIME_MODE_KEY) {
+      const newBounds = {
+        start: timestamp + this.offsets.start,
+        end: timestamp + this.offsets.end
+      };
 
-    this.boundsVal = newBounds;
-    this.emit('bounds', this.boundsVal, true);
+      this.boundsVal = newBounds;
+      // "bounds" will be deprecated in a future release
+      this.emit('bounds', this.boundsVal, true);
+      this.emit(TIME_CONTEXT_EVENTS.boundsChanged, this.boundsVal, true);
+    }
   }
 
   /**
-   * Checks if this time context is in real-time mode or not.
+   * Get the time system of the TimeAPI.
+   * @returns {TimeSystem} The currently applied time system
+   * @memberof module:openmct.TimeAPI#
+   * @method getTimeSystem
+   */
+  getTimeSystem() {
+    return this.system;
+  }
+
+  /**
+   * Set the time system of the TimeAPI.
+   * @param {TimeSystem | string} timeSystemOrKey
+   * @param {module:openmct.TimeAPI~TimeConductorBounds} bounds
+   * @fires module:openmct.TimeAPI~timeSystem
+   * @returns {TimeSystem} The currently applied time system
+   * @memberof module:openmct.TimeAPI#
+   * @method setTimeSystem
+   */
+  setTimeSystem(timeSystemOrKey, bounds) {
+    if (!this.isRealTime() && !bounds) {
+      throw new Error('Must specify bounds when changing time system without an active clock.');
+    }
+
+    if (timeSystemOrKey === undefined) {
+      throw 'Please provide a time system';
+    }
+
+    let timeSystem;
+
+    if (typeof timeSystemOrKey === 'string') {
+      timeSystem = this.timeSystems.get(timeSystemOrKey);
+
+      if (timeSystem === undefined) {
+        throw `Unknown time system ${timeSystemOrKey}. Has it been registered with 'addTimeSystem'?`;
+      }
+    } else if (typeof timeSystemOrKey === 'object') {
+      timeSystem = timeSystemOrKey;
+
+      if (!this.timeSystems.has(timeSystem.key)) {
+        throw `Unknown time system ${timeSystemOrKey.key}. Has it been registered with 'addTimeSystem'?`;
+      }
+    } else {
+      throw 'Attempt to set invalid time system in Time API. Please provide a previously registered time system object or key';
+    }
+
+    this.system = this.#copy(timeSystem);
+    /**
+     * The time system used by the time
+     * conductor has changed. A change in Time System will always be
+     * followed by a bounds event specifying new query bounds.
+     *
+     * @event module:openmct.TimeAPI~timeSystem
+     * @property {TimeSystem} The value of the currently applied
+     * Time System
+     * */
+    this.emit(TIME_CONTEXT_EVENTS.timeSystemChanged, this.#copy(this.system));
+
+    if (bounds) {
+      this.setBounds(bounds);
+    }
+
+    return this.system;
```

Comment:
Why are we returning anything from setter functions?

Reply:
Will update!